apps/native/app/src/screens/inbox/inbox.tsx

```diff
@@ -675,6 +844,113 @@ export const InboxScreen: NavigationFunctionComponent<{
           ) : null
         }
       />
+      {selectState && selectedItems.length ? (
+        <ActionBar
+          loading={bulkSelectActionLoading}
+          onClickStar={async () => {
+            const result = await bulkSelectActionMutation({
+              variables: {
+                input: { action: 'bookmark', documentIds: selectedItems },
+              },
+            })
+            if (result.data?.postMailActionV2?.success) {
+              // If success, update selected documents to be starred in the cache
+              selectedItems.forEach((id) => {
+                client.cache.modify({
+                  id: client.cache.identify({ __typename: 'DocumentV2', id }),
+                  fields: {
+                    bookmarked: () => true,
+                  },
+                })
+              })
+              resetSelectState()
+              showToastForBulkSelectAction({
+                variant: 'success',
+                title: intl.formatMessage({
+                  id: 'inbox.bulkSelect.starSuccess',
+                }),
+              })
+            } else {
+              showToastForBulkSelectAction({
+                variant: 'success',
+                title: intl.formatMessage({
+                  id: 'inbox.bulkSelect.starError',
+                }),
+                message: intl.formatMessage({
+                  id: 'inbox.bulkSelect.pleaseTryAgain',
+                }),
+              })
+            }
+          }}
+          onClickArchive={async () => {
+            const result = await bulkSelectActionMutation({
+              variables: {
+                input: { action: 'archive', documentIds: selectedItems },
+              },
+            })
+            if (result.data?.postMailActionV2?.success) {
+              resetSelectState()
+              res.refetch()
+              showToastForBulkSelectAction({
+                variant: 'success',
+                title: intl.formatMessage({
+                  id: 'inbox.bulkSelect.archiveSuccess',
+                }),
+              })
+            } else {
+              showToastForBulkSelectAction({
+                variant: 'success',
```

Comment:
~Should we use another variant for errors?

Reply:
Great catch!